plugwise/smile.py

```diff
@@ -457,26 +463,26 @@ async def async_update(self):
 
         return [self.gw_data, self.gw_devices]
 
-    async def _set_schedule_state_legacy(self, name, status):
+    async def _set_schedule_state_legacy(self, name: str, status: str) -> bool:
         """Helper-function for set_schedule_state()."""
-        schema_rule_id = None
+        schema_rule_id: str | None = None
         for rule in self._domain_objects.findall("rule"):
             if rule.find("name").text == name:
                 schema_rule_id = rule.attrib["id"]
 
         if schema_rule_id is None:
             return False
 
-        template_id = None
-        state = "false"
-        if str(status) == "on":
+        template_id: str = None
```

Comment:
Template isn't a string in this case? Its `None`.

Reply:
str | None is the solution then?

Or is it better to init as an empty string: ""?

Reply:
I'll just remove the line. That seems to work as well.